File: Python/Task27/n27424.py
# https://inf.reshuege.ru/problem?id=27424

f = open('./Data/27-B_demo.txt')
n = int(f.readline())
mind = 20001
sum = 0
for i in range(n):
	x, y = map(int, f.readline().split())
	sum += max(x, y)
	d = abs(x - y)
	if d % 3 != 0:
		mind = min(mind, d)

if sum % 3 != 0:
	print(sum)
else:
	print(sum - mind)

File: C++/Task27/n27424/n27424.cpp
// https://inf.reshuege.ru/problem?id=27424

#include <iostream>
#include <fstream>
#include <algorithm>
#include <cmath>

int main()
{
	std::fstream myfile(".\\Data\\27-A_demo.txt", std::ios_base::in);
	int n;
	myfile >> n;
	int x, y, d, min = 20001, sum = 0;
	for (int i = 0; i < n; i++)
	{
		myfile >> x >> y;
		sum += std::max(x, y);
		d = std::abs(x - y);
		if (d % 3 != 0)
		{
			min = std::min(min, d);
		}
	}
	int result = sum % 3 != 0 ? sum : sum - min;
	std::cout << result << std::endl;
	return 0;
}
